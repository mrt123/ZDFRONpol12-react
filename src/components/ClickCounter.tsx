// 1 Komponent ma wyswietlic cyfre 0
// 2 Komponent ma wyświetlic cyfrę 0 ale źródłem tej cyfry ma byc stan
// 3 Komponent ma wyśwuetlić button oprócz cyfry
// 4 gdy klinkniemy button cyfra w stanie ma zwiekszyc sie o 1
// a) zdefiniuj funkcję która zmienia stan
// b) gdy kliknę na button, alert "działa"
// c) wypełniam założenia pkt. 4

import React, { useState } from "react";
const ClickCounter = () => {
  const [count, setCount] = useState(0);
  const increaseCount = () => {
    setCount(count + 1);
  };
  return <button onClick={increaseCount}>You clicked me {count} times</button>;
};
export default ClickCounter;
