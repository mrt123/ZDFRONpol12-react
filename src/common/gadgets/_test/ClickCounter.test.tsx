import { fireEvent, render, screen } from "@testing-library/react";
import ClickCounter from "../ClickCounter";

describe("ClickCounter", () => {
  it("matches snapshot", () => {
    const { container } = render(<ClickCounter />);
    expect(container).toMatchSnapshot();
  });

  it("clicking 1 time increases number by 1", () => {
    const { container, getByText } = render(<ClickCounter />);

    const elementWIthExpectedTextBeforeClick = getByText("Click me 0");
    expect(elementWIthExpectedTextBeforeClick).toBeInTheDocument();

    fireEvent.click(screen.getByText(/click me/i));

    const elementWithExpectedTextAfterClick = getByText("Click me 1");
    expect(elementWithExpectedTextAfterClick).toBeInTheDocument();
  });

  // TODO: test what happens after 10 clicks
});
