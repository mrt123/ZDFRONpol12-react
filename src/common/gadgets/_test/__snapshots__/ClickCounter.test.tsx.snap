// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickCounter matches snapshot 1`] = `
<div>
  <button
    style="margin: 2px; font-size: 10px;"
  >
    Click me 
    0
  </button>
</div>
`;
